File: CodeWars/JavaScript/6Kyu/BallUpwards.js
function maxBall(v0) {
  speedConverted = v0/3.6;
  heights = [];
  
  for(let i = 0; i<=speedConverted/2; i=i+0.1){
    height = speedConverted*(i) - 0.5*9.81*(i)*(i); 
    
    heights.push(height);
  }
  
  return heights.indexOf(Math.max( ...heights ));
}

console.log(maxBall(15));

File: CodeWars/Java/6Kyu/BallUpwards.java
import java.util.*;

public class BallUpwards {
  public static int maxBall(int v0) {
    Double speedConverted = v0 / 3.6;
    List<Double> heights = new ArrayList<Double>();

    for (double i = 0; i <= speedConverted / 2; i = i + 0.1) {
      Double height = speedConverted * (i) - 0.5 * 9.81 * (i) * (i);

      heights.add(height);
    }

    return heights.indexOf(Collections.max(heights));
  }

  public static void main(String[] args) {
    System.out.println(maxBall(15));
  }
}